src/Migrations/MigrationCreator.php

```diff
@@ -196,18 +196,28 @@ private function createMigration($name, $columnData)
      *
      * @return string
      */
-    private function createForeignKeyMigration($upData, $downData)
+    private function createForeignKeyMigration($upData, $downData, $schemaName = null)
     {
         // Update time interval each time so that migration
         // files are created with different timestamp.
         $this->setTimeInterval($this->getTimeInterval() + 60);
 
-        $path = $this->getDatePrefix() . '_create_foreign_keys_table.php';
+        $schemaPart = $classPart = '';
+
+        if ($schemaName) {
+            $schemaPart = '_for_' . str_replace(' ', '_', $schemaName);
```

Comment:
Maybe we need to convert `$schemaName` to lowercase here in case anyone provides uppercase name.